stumpy/mmotifs.py

```diff
@@ -0,0 +1,187 @@
+
+    motif_indices: numpy.ndarray
+        The indices corresponding to a set of subsequences matches for each motif.
+
+    motif_subspaces: numpy.ndarray
+        A numpy.ndarray consisting of arrays that contain the `k`-dimensional
+        subspace for each motif.
+
+    motif_mdls: numpy.ndarray
+        A numpy.ndarray consisting of arrays that contain the mdl results for
+        finding the dimension of each motif
+    """
+    if max_motifs < 1:  # pragma: no cover
+        logger.warning(
+            "The maximum number of motifs, `max_motifs`, "
+            "must be greater than or equal to 1"
+        )
+        logger.warning("`max_motifs` has been set to `1`")
+        max_motifs = 1
+
+    # Calculate subsequence length
+    T = core._preprocess(T)  # convert dataframe to array if necessary
+    m = T.shape[-1] - P.shape[-1] + 1
+
+    # Calculate exclusion zone
+    excl_zone = int(np.ceil(m / config.STUMPY_EXCL_ZONE_DENOM))
+
+    # Precompute rolling means and standard deviations
+    T, M_T, Σ_T = core.preprocess(T, m)
+
+    if max_matches is None:
+        max_matches = np.inf
+
+    motif_distances = []
+    motif_indices = []
+    motif_subspaces = []
+    motif_mdls = []
+
+    # Identify the first multidimensional motif as candidate motif and get its
+    # nearest neighbor
+    candidate_idx = np.argmin(P, axis=1)
+    nn_idx = I[np.arange(len(candidate_idx)), candidate_idx]
+
+    # Iterate over each motif
+    while len(motif_distances) < max_motifs:
+
+        # Choose dimension k using MDL
+        mdls, subspaces = mdl(T, m, candidate_idx, nn_idx)
+        k = np.argmin(mdls)
+        motif_mdls.append(mdls)
+
+        # Calculate default cutoff value
+        if cutoff is None:
+            P_copy = P.copy().astype(np.float64)
+            P_copy[np.isinf(P_copy)] = np.nan
+            P_copy = P_copy[k, :]
+            cutoff_var = np.nanmax(
+                [np.nanmean(P_copy - 2.0 * np.nanstd(P_copy)), np.nanmin(P_copy)]
+            )
+            cutoff = None
+        else:
+            if isinstance(cutoff, np.ndarray) or isinstance(cutoff, list):
+                cutoff_var = cutoff[k]
+            else:
+                cutoff_var = cutoff
+
+        # Get k-dimensional motif
+        motif_idx = candidate_idx[k]
+        motif_value = P[k, motif_idx]
+        if motif_value > cutoff_var or not np.isfinite(motif_value):
+            break
+
+        # Compute subspace and find related dimensions
+        motif_subspaces.append(subspaces[k])
+        sub_dims = T[subspaces[k]]
+
+        # Stop iteration if max_distance is a constant and the k-dim.
+        # matrix profile value is larger than the maximum distance.
+        if isinstance(max_distance, float) and motif_value > max_distance:
+            break
+
+        # Get multidimensional Distance Profile to find the max_matches
+        # nearest neighbors
+        M_T_sub = M_T[subspaces[k]]

```

Comment:
These two lines don't seem necessary. I think it makes `query_matches` better to see:

```
M_T=M_T[subspaces[k]]
Σ_T=Σ_T[subspaces[k]]
```

Because the reader doesn't have to follow "what is inside of `M_T_sub`?" 

The same goes for `T=T[subspaces[k]]` and `Q=T[subspaces[k]], motif_idx : motif_idx + m`

I think this actually makes it more obvious what is being passed to `query_match` rather than having the reader need to scroll up to keep track